modules/core/02-client/keeper/client_test.go

```diff
@@ -42,216 +42,186 @@ func (suite *KeeperTestSuite) TestCreateClient() {
-				NextValidatorsHash: suite.valSetHash,
+				NextValidatorsHash: suite.chainB.Vals.Hash(),
 			}
-			suite.keeper.SetClientConsensusState(suite.ctx, clientID, height1, prevConsState)
+			suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientConsensusState(suite.chainA.GetContext(), clientID, height1, prevConsState)
 
-			height2 := types.NewHeight(0, 2)
-
-			// store intermediate consensus state to check that trustedHeight does not need to be hightest consensus state before header height
-			intermediateConsState := &ibctmtypes.ConsensusState{
+			height5 := types.NewHeight(0, 5)
+			// store next consensus state to check that trustedHeight does not need to be hightest consensus state before header height
+			nextConsState := &ibctmtypes.ConsensusState{
 				Timestamp:          suite.past.Add(time.Minute),
-				NextValidatorsHash: suite.valSetHash,
+				NextValidatorsHash: suite.chainB.Vals.Hash(),
 			}
-			suite.keeper.SetClientConsensusState(suite.ctx, clientID, height2, intermediateConsState)
+			suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientConsensusState(suite.chainA.GetContext(), clientID, height5, nextConsState)
 
+			height3 := types.NewHeight(0, 3)
 			// updateHeader will fill in consensus state between prevConsState and suite.consState
 			// clientState should not be updated
-			updateHeader = createPastUpdateFn(suite)
+			updateHeader = createPastUpdateFn(height3, height1)
 			// set conflicting consensus state in store to create misbehaviour scenario
 			conflictConsState := updateHeader.ConsensusState()
 			conflictConsState.Root = commitmenttypes.NewMerkleRoot([]byte("conflicting apphash"))
-			suite.keeper.SetClientConsensusState(suite.ctx, clientID, updateHeader.GetHeight(), conflictConsState)
-			return nil
+			suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientConsensusState(suite.chainA.GetContext(), clientID, updateHeader.GetHeight(), conflictConsState)
 		}, true, true},
-		{"misbehaviour detection: monotonic time violation", func() error {
-			clientState = ibctmtypes.NewClientState(testChainID, ibctmtypes.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, testClientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath, false, false)
-			clientID, err = suite.keeper.CreateClient(suite.ctx, clientState, suite.consensusState)
+		{"misbehaviour detection: monotonic time violation", func() {
+			clientState := path.EndpointA.GetClientState().(*ibctmtypes.ClientState)
+			clientID := path.EndpointA.ClientID
+			trustedHeight := clientState.GetLatestHeight().(types.Height)
 
 			// store intermediate consensus state at a time greater than updateHeader time
 			// this will break time monotonicity
-			incrementedClientHeight := testClientHeight.Increment().(types.Height)
+			incrementedClientHeight := clientState.GetLatestHeight().Increment().(types.Height)
 			intermediateConsState := &ibctmtypes.ConsensusState{
 				Timestamp:          suite.header.Header.Time.Add(2 * time.Hour),
```

Comment:
`suite.header.Header.Time.Add(2 * time.Hour)` should be `suite.coordinator.CurrentTime.Add(2 * time.Hour)`